spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/couchbase/CouchbaseConfiguration.java

```diff
@@ -57,7 +57,14 @@ public DefaultCouchbaseEnvironment couchbaseEnvironment() {
 	@Bean
 	@Primary
 	public Cluster couchbaseCluster() {
-		return CouchbaseCluster.create(couchbaseEnvironment(), determineBootstrapHosts());
+		CouchbaseCluster couchbaseCluster = CouchbaseCluster
+				.create(couchbaseEnvironment(), determineBootstrapHosts());
+		if (this.properties.getUsername() != null
+				&& this.properties.getPassword() != null) {
```

Comment:
The `username` and `password` properties are initialized to empty string (`""`). Will they ever be null? A better check might be `.isEmpty()`.

Reply:
Opps how can ı miss 👎 Thanks for your awesome attention